components/builder-api/src/server/resources/profile.rs

```diff
@@ -268,18 +271,18 @@ async fn set_license(req: HttpRequest,
                 }
             };
 
-            let expiration_str = json["entitlements"].as_array().and_then(|entitlements| {
-                                                                    entitlements
-                        .iter()
-                        .find_map(|ent| {
-                            if ent["name"] == "Habitat" {
-                                ent["period"]["end"].as_str()
-                            } else {
-                                None
-                            }
-                        })
-                        .or_else(|| entitlements.first()?.get("period")?.get("end")?.as_str())
-                                                                });
+            let today = chrono::Utc::now().date_naive();
+
+            let expiration_str =
+                json["entitlements"].as_array().and_then(|entitlements| {
+                                                   entitlements.iter().find_map(|ent| {
+                        let end_str = ent.get("period")?.get("end")?.as_str()?;
+                        match NaiveDate::parse_from_str(end_str, "%Y-%m-%d") {
+                            Ok(end_date) if end_date >= today => Some(end_str),
+                            _ => None
+                        }
+                    })
+                                               });
```

Comment:
We should throw an error message stating that all entitlements are expired rather than stating we were unable to determine the expiration date.